pkg/kubectl/cmd/config/use_context.go

```diff
@@ -82,10 +84,16 @@ func (o *useContextOptions) complete(cmd *cobra.Command) bool {
 	return true
 }
 
-func (o useContextOptions) validate() error {
+func (o useContextOptions) validate(config *clientcmdapi.Config) error {
 	if len(o.contextName) == 0 {
 		return errors.New("You must specify a current-context")
 	}
 
-	return nil
+	for _, context := range config.Contexts {
+		if context.Cluster == o.contextName {
```

Comment:
Ah right, I vaguely remember the kerfuffle over serializing maps. Thanks for the explanation.

@mikedanese, tl;dr:  validate should do
```
for key, _ := range config.Contexts {
		if key == o.contextName {
				return nil
```